prbt_hardware_support/test/integrationtests/integrationtest_execute_brake_test.cpp

```diff
@@ -0,0 +1,147 @@
+
+static constexpr double WAIT_FOR_BRAKE_TEST_SERVICE_TIMEOUT_S{5.0};
+static constexpr uint16_t MODBUS_BRAKE_TEST_PREPARE_VALUE {0};
+static constexpr uint16_t MODBUS_BRAKE_TEST_EXPECTED_VALUE {1};
+
+/**
+ * @tests{Execute_BrakeTest_mechanism,
+ *  Test the BrakeTest service node.
+ * }
+ *
+ * Test Sequence:
+ *  0. Preparing modbus mock
+ *  1. Initialize CANOpen mock, JointStatesPublisher mock and manipulator mock
+ *  2. Wait for BrakeTest service
+ *  3. Make the service call
+ *  4. Check contents of relevant modbus registers
+ *  5. Shutdown of modbus mock
+ *
+ * Expected Results:
+ *  0. -
+ *  1. -
+ *  2. Service is available
+ *  3. Service call is successful
+ *  4. Brake test success was set in modbus
+ *  5. -
+ */
+TEST(IntegrationtestExecuteBrakeTest, testBrakeTestService)
+{
+  using namespace prbt_hardware_support;
+  ros::NodeHandle nh;
+  ros::NodeHandle nh_private("~");
+
+  /**********
+   * Step 0 *
+   **********/
+  std::string ip;
+  int port;
+  ASSERT_TRUE(nh_private.getParam("modbus_server_ip", ip));
+  ASSERT_TRUE(nh_private.getParam("modbus_server_port", port));
+
+  ModbusApiSpec write_api_spec {nh, "write_api_spec/"};
+  unsigned int modbus_register_size {write_api_spec.getMaxRegisterDefinition() + 1U};
+  prbt_hardware_support::PilzModbusServerMock modbus_server(modbus_register_size);
+  std::thread modbus_server_thread( &initalizeAndRun<prbt_hardware_support::PilzModbusServerMock>,
+                                    std::ref(modbus_server), ip.c_str(), static_cast<unsigned int>(port) );
+  ASSERT_TRUE(write_api_spec.hasRegisterDefinition(modbus_api_spec::BRAKETEST_PERFORMED));
+  unsigned int register_perf = write_api_spec.getRegisterDefinition(modbus_api_spec::BRAKETEST_PERFORMED);
+  ASSERT_TRUE(write_api_spec.hasRegisterDefinition(modbus_api_spec::BRAKETEST_RESULT));
+  unsigned int register_res = write_api_spec.getRegisterDefinition(modbus_api_spec::BRAKETEST_RESULT);
+
+  modbus_server.setHoldingRegister({{register_perf, MODBUS_BRAKE_TEST_PREPARE_VALUE}});
+  modbus_server.setHoldingRegister({{register_res, MODBUS_BRAKE_TEST_PREPARE_VALUE}});
+
+  /**********
+   * Step 1 *
+   **********/
+  CANOpenChainNodeMock canopen_mock;
+
+  JointStatesPublisherMock joint_states_pub;
+  joint_states_pub.startAsync();
+
+  ManipulatorMock manipulator;
+  manipulator.advertiseHoldService(nh, CONTROLLER_HOLD_MODE_SERVICE_NAME);
+  manipulator.advertiseUnholdService(nh, CONTROLLER_UNHOLD_MODE_SERVICE_NAME);
+
+  /**********
+   * Step 2 *
+   **********/
+  ros::ServiceClient brake_test_srv_client_ = nh.serviceClient<BrakeTest>(EXECUTE_BRAKE_TEST_SERVICE_NAME);
+  EXPECT_TRUE(brake_test_srv_client_.waitForExistence(ros::Duration(WAIT_FOR_BRAKE_TEST_SERVICE_TIMEOUT_S)));
+
+  /**********
+   * Step 3 *
+   **********/
+  BrakeTest srv;
+  EXPECT_TRUE(brake_test_srv_client_.call(srv));
+  ros::Duration(1).sleep(); // make sure values are set in modbus mock
```

Comment:
Please avoid sleep at any costs. This also seems redundant with your next lines?